### ISSUE TITLE
Build failure, missing JSON.h?

### ISSUE THREAD

**Issue by chmorgan** [nkolban/ESP32_Explorer#24]
Using latest esp-idf master and ESP32_Explorer as of today.

```
$ make
CXX build/main/BLEExplorer.o
In file included from /Users/cmorgan/projects/ESP32_Explorer/main/./BLEExplorer.cpp:8:0:
/Users/cmorgan/projects/ESP32_Explorer/main/./BLEExplorer.h:11:18: fatal error: JSON.h: No such file or directory
 #include "JSON.h"
                  ^
compilation terminated.
make[1]: *** [BLEExplorer.o] Error 1
```

**Comment by chegewara** [comment]
@chmorgan You need to add this library to components (whole cpp_utils):
https://github.com/nkolban/esp32-snippets/tree/master/cpp_utils

**Comment by chmorgan** [comment]
Not using a submodule to pull that in?

**Comment by chegewara** [comment]
No, its just symlink from our dev environment. You need to do this manually by copying folder or with symlink.

**Comment by chmorgan** [comment]
Ahh. Submodule would make that automatic for users :-)

I'll pull it in and try rebuilding, thanks for the pointer.

**Comment by chmorgan** [comment]
Things are building now.

I'm trying to see how much space nghttp2 is taking up but I'm having trouble seeing it apart from the rest of the code. It looks like it isn't built in as a component. I'm trying to do a size comparison of nghttp2 vs. libesphttpd to see if I should switch http servers.

**Comment by chegewara** [comment]
About this stuff i cant be much helpfull. But im pretty sure mr @nkolban will know how to answer it.
